planner/core/task.go

```diff
@@ -473,32 +695,64 @@ func (p *basePhysicalAgg) newPartialAggregate() (partial, final PhysicalPlan) {
 
 func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 	t := tasks[0].copy()
+	inputRows := t.count()
 	if cop, ok := t.(*copTask); ok {
 		partialAgg, finalAgg := p.newPartialAggregate()
 		if partialAgg != nil {
 			if cop.tablePlan != nil {
+				cop.finishIndexPlan()
 				partialAgg.SetChildren(cop.tablePlan)
 				cop.tablePlan = partialAgg
 			} else {
 				partialAgg.SetChildren(cop.indexPlan)
 				cop.indexPlan = partialAgg
 			}
+			cop.addCost(p.GetCost(inputRows, false))
 		}
 		t = finishCopTask(p.ctx, cop)
+		inputRows = t.count()
 		attachPlan2Task(finalAgg, t)
 	} else {
 		attachPlan2Task(p, t)
 	}
-	t.addCost(t.count() * cpuFactor)
-	if p.hasDistinctFunc() {
-		t.addCost(t.count() * cpuFactor * distinctAggFactor)
-	}
+	t.addCost(p.GetCost(inputRows, true))
 	return t
 }
 
+// GetCost computes cost of stream aggregation considering CPU/memory.
+func (p *PhysicalStreamAgg) GetCost(inputRows float64, isRoot bool) float64 {
+	var cpuCost float64
+	if isRoot {
+		cpuCost = inputRows * cpuFactor * float64(len(p.AggFuncs))
+	} else {
+		cpuCost = inputRows * copCPUFactor * float64(len(p.AggFuncs))
+	}
+	rowsPerGroup := inputRows / p.statsInfo().RowCount
```

Comment:
Can we guarantee that `RowCount` is not 0?

Reply:
`getCardinality` is updated to guarantee that the estimated output row count of aggregation would not be smaller than 1.0